pkg/startup/bootstrap.go

```diff
@@ -0,0 +1,63 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2017-2018 Canonical Ltd
+// Copyright (C) 2018 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package startup
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/edgexfoundry/device-sdk-go"
+	"github.com/edgexfoundry/device-sdk-go/internal/common"
+	ds_models "github.com/edgexfoundry/device-sdk-go/pkg/models"
+)
+
+// Bootstrap the Device Service in a default way
+func Bootstrap(serviceVersion string, driver ds_models.ProtocolDriver) {
+	var confProfile string
+	var confDir string
+
+	//flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError) // clean up existing flag defined by other code
+	flag.BoolVar(&common.UseRegistry, "registry", false, "Indicates the service should use the registry.")
+	flag.BoolVar(&common.UseRegistry, "r", false, "Indicates the service should use registry.")
```

Comment:
Directly setting an internal var like this means that it's not even possible to use device.NewService without using the boostrap package! Usage of registry should be a parameter to NewService like it was previously.

Reply:
Sorry, I missed it.  Added it back now.